lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1736,6 +1855,46 @@ instance FromText PostExternalTransactionData where
         Right load ->
             pure $ PostExternalTransactionData load
 
+instance FromText ApiPubKey where
+    fromText txt = case detectEncoding (T.unpack txt) of
```

Comment:
If the only possible encoding is bech32, then it would be best to just decode in bech32 without trying to detect encoding.

Reply:
agree, but if someone gives base16 or base58 then one will be given the error due to this https://github.com/input-output-hk/cardano-addresses/blob/master/core/lib/Codec/Binary/Encoding.hs#L100 and if bech32 format is detected then data part, hrp, etc are later verified